tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -90,6 +92,29 @@ def test_read_config_valid(self):
         assert 'foo' in kp.bindings
         assert 'ctrl+x' in kp.special_bindings
 
+    def test_on_keyconfig_changed(self):
```

Comment:
I'd prefer if this was split into two tests - one for the `AssertionError` (which doesn't need the mock and all), and one for the rest.